holidays/countries/fiji.py

```diff
@@ -0,0 +1,182 @@
+        HinduCalendarHolidays.__init__(self, cls=FijiHinduHolidays, show_estimated=True)
+        InternationalHolidays.__init__(self)
+        IslamicHolidays.__init__(self, cls=FijiIslamicHolidays, show_estimated=True)
+        StaticHolidays.__init__(self, FijiStaticHolidays)
+        kwargs.setdefault("observed_rule", SAT_SUN_TO_NEXT_MON)
+        super().__init__(*args, **kwargs)
+
+    def _populate_public_holidays(self):
+        # New Year's Day.
+        self._add_observed(self._add_new_years_day("New Year's Day"))
+
+        # Good Friday.
+        self._add_good_friday("Good Friday")
+
+        # Easter Saturday.
+        self._add_holy_saturday("Easter Saturday")
+
+        # Easter Monday.
+        self._add_easter_monday("Easter Monday")
+
+        # National Sports Day.
+        if self._year <= 2018:
+            self._add_holiday_last_fri_of_jun("National Sports Day")
+
+        if self._year <= 2022:
+            # Constitution Day.
+            self._add_observed(self._add_holiday_sep_7("Constitution Day"))
+
+        if self._year >= 2023:
+            # Girmit Day.
+            self._move_holiday(
+                self._add_holiday_may_14("Girmit Day"),
+                rule=ALL_TO_NEAREST_MON,
+                show_observed_label=False,
+            )
+
+        # Ratu Sir Lala Sukuna Day.
+        name = "Ratu Sir Lala Sukuna Day"
+        if self._year == 2023:
+            self._add_holiday_last_mon_of_may(name)
+        elif self._year >= 2024:
+            self._add_holiday_last_fri_of_may(name)
+
+        # Fiji Day.
+        self._add_holiday_oct_10("Fiji Day")
+
+        # Prophet Mohammed's Birthday
+        self._populate_observed(self._add_mawlid_day("Prophet Mohammed's Birthday"))
+
+        # Diwali
+        self._add_observed(self._add_diwali_india("Diwali"))
+
+        # Christmas Day.
+        self._add_observed(self._add_christmas_day("Christmas Day"), rule=SAT_SUN_TO_NEXT_MON_TUE)
+
+        # Boxing Day.
+        self._add_observed(self._add_christmas_day_two("Boxing Day"), rule=SAT_SUN_TO_NEXT_MON_TUE)
+
+    def _populate_workday_holidays(self):
+        if self._year >= 2023:
+            # Constitution Day.
+            self._add_holiday_sep_7("Constitution Day")
+
+
+class FJ(Fiji):
+    pass
+
+
+class FJI(Fiji):
+    pass
+
+
+class FijiHinduHolidays(_CustomHinduHolidays):
+    # https://www.timeanddate.com/holidays/fiji/diwali
+    DIWALI_INDIA_DATES = {
+        2016: (OCT, 31),
+        2017: (OCT, 19),
+        2018: (NOV, 7),
+        2019: (OCT, 28),
+        2020: (NOV, 14),
+        2021: (NOV, 4),
+        2022: (OCT, 25),
+        2023: (NOV, 13),
+        2024: (NOV, 1),
+        2025: (OCT, 21),
+    }
+
+
+class FijiIslamicHolidays(_CustomIslamicHolidays):
+    MAWLID_DATES = {
+        2019: (NOV, 11),
+        2020: (NOV, 2),
+        2021: (OCT, 18),
+        2022: (OCT, 7),
+        2023: (OCT, 2),
+        2024: (SEP, 16),
+        2025: (SEP, 8),
```

Comment:
```suggestion
        2016: (DEC, 12),
        2017: (DEC, 2),
        2018: (NOV, 19),
        2019: (NOV, 9),
        2020: (OCT, 31),
        2021: (OCT, 18),
        2022: (OCT, 7),  # looks like observed on FRI
        2023: (SEP, 30),
        2024: (SEP, 16),
        2025: (SEP, 6),
```
The dates are not 100% precise, but the annual offsets are more realistic for the Islamic calendar.